app/controllers/assessment_questions_controller.rb

```diff
@@ -11,9 +11,39 @@ def your_risk_post
 
     end
 
-    (session['current_assessment'] ||= Hash.new)[:confidence_level_required] = params[:confidence_level_required]
+    get_assessment()['confidence_level_required'] = params[:confidence_level_required]
 
     redirect_to controller: 'assessments', action: 'overview'
   end
 
+  def choose_evidence_get
+    render "assessments/choose-evidence"
+  end
+
+  def choose_evidence_post
+    if (not params[:evidence_type]) && (params[:evidence_type_other].blank?)
+      @errors[:evidence_type] = 'You must choose a piece of evidence'
+      render "assessments/choose-evidence" and return
+    end
+    if params[:id] == "new"
+      id = SecureRandom.uuid
+    else
+      id = params[:id]
+    end
+    if params[:choose_evidence] == "other"
+      params[:evidence_type] = nil
+    else
+      params[:evidence_type_other] = nil
+    end
+    get_evidence(id)['evidence_type'] = params[:evidence_type]
+    get_evidence(id)['evidence_type_other'] = params[:evidence_type_other]
```

Comment:
Is this a rails limitation that means we're using nested hashes? It seems as if some of this could be classes instead? That might allow us to more easily enforce the restriction of an evidence either having a defined type or being other?

Reply:
Well we could use some kind of normal model class, definitely considered it and I'm not fully confident we made the right choice.

But leaning in favour of this approach:

 - it seems to be strongly recommended to store simple things like strings and hashes in the session, rather than more complex objects, I'm worried about getting into a world of serialisation pain that I don't understand
 - I'm thinking that the right place to store the questions and answers is potentially a yaml file. This seems a common pattern, and the other options of storing lookup hashes in a model, or having separate models for lookup tables, both would add complexity (particularly as we aren't planning to use a database at this stage).

Further objections to the approach I'm taking:

-  I must admit the previous systems I worked on here (looking at you Digital Marketplace) I didn't always find the "data-file plus hashes everywhere" approach particularly fun to work with.
- why do we want to use the session like this? I've always thought it better practice to use a database [answer: this lifts off a whole bunch of security/access-control concerns in building an MVP, seeing as long-term storage isn't a requirement]

Eh.

Reply:
Fine. I've also had bad experiences with an application that over-used "data-file + hashes".

But I'm happy if this is a deliberate choice.